src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,359 @@
+        // All of these created notifications should be returned
+        for (int i = 0; i < numOfExpectedOngoingNotification; i += 2) {
+            n = getNewNotificationAttributes();
+            n.setStartTime(Instant.now().minusSeconds(3600));
+            n.setEndTime(Instant.now().plusSeconds(3600));
+            n.setTargetUser(expectedTargetUser);
+            expectedIds.add(notificationsDb.createEntity(n).getNotificationId());
+
+            n = getNewNotificationAttributes();
+            n.setStartTime(Instant.now().minusSeconds(3600));
+            n.setEndTime(Instant.now().plusSeconds(3600));
+            n.setTargetUser(NotificationTargetUser.GENERAL);
+            expectedIds.add(notificationsDb.createEntity(n).getNotificationId());
+        }
+
+        List<NotificationAttributes> retrieved = notificationsDb.getActiveNotificationsByTargetUser(expectedTargetUser);
+
+        // Verifies that what was retrieved is exactly the same as what was expected.
+        this.verifyNotifications(retrieved, expectedIds);
+    }
+
+    @Test
+    public void testCreateNotification() throws Exception {
+        ______TS("typical success case");
+        NotificationAttributes n1 = createNewNotification();
+
+        ______TS("failure: duplicate notification with the same ID");
+        assertThrows(EntityAlreadyExistsException.class, () -> notificationsDb.createEntity(n1));
+
+        ______TS("failure: invalid non-null parameters");
+        NotificationAttributes n2 = getNewNotificationAttributes();
+        n2.setTitle("");
+        assertThrows(InvalidParametersException.class, () -> notificationsDb.createEntity(n2));
+
+        ______TS("failure: null parameters");
+        assertThrows(AssertionError.class, () -> notificationsDb.createEntity(null));
+    }
+
+    // TODO: for extension, some fields are not allowed to be updated after shown is true
+    @Test
+    public void testUpdateNotification() throws Exception {
+        ______TS("failure: update non-existent notification");
+        assertThrows(EntityDoesNotExistException.class, () ->
+                notificationsDb.updateNotification(NotificationAttributes.updateOptionsBuilder(NON_EXISTENT_ID)
+                        .withTitle("title")
+                        .build()));
+
+        ______TS("failure: invalid non-null parameters");
+        // Empty title is used here, which triggers InvalidParametersException
+        assertThrows(InvalidParametersException.class, () ->
+                notificationsDb.updateNotification(NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withTitle("")
+                        .build()));
+
+        ______TS("failure: null update options");
+        assertThrows(AssertionError.class, () -> notificationsDb.updateNotification(null));
+    }
+
+    @Test
+    public void testUpdateNotification_bulkUpdate_shouldUpdateSuccessfully() throws Exception {
+        ______TS("success: bulk update");
```

Comment:
It seems to me that this test can be part of `testUpdateNotification`, like a "standard success case", not sure what do you mean by bulk update.

Reply:
This wording comes from `AccountsDbTest`. I will merge the two together.